### PR TITLE
[Wallet] Transaction details screen

### LINKED ISSUE #1775: Wallet - data-bindings for transaction details
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to see transaction details when touching the right arrow icon next to a transaction.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

*Summary*: 
Transaction details should contain  the following informations

- [ ] value
- [ ] date
- [ ] block
- [ ] from
- [ ] to
- [ ] gas limit
- [ ] gas price
- [ ] gas used
- [ ] nonce
- [ ] data

- [ ] number of confirmations

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

- [x] modify status-im.utils.transactions/get-transactions so that the fetched transactions have complete informations 
- [ ] for the number of confirmations the subscription to transaction details should include the current block and compute the difference between the transaction block and the current block, the result should be accessible in :confirmations key of the subscription

### PR DESCRIPTION
**Pull request by yenda**
fixes #1775

### Summary:
- Add transaction details screen for new wallet
- change transaction views from modal to screens

### Steps to test:
- Open Status
- Go to wallet
- Go to transactions list
- Go to transaction history and click the right arrow icon next to a transaction

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 1]
```diff
@@ -88,3 +88,74 @@
...
+   :flex-direction  :column
+   :margin-left     12
+   :margin-vertical 7})
+
+(def transaction-details-header-value
+  {:font-size   16
+   :color       styles/color-black})
+
+(def transaction-details-header-date
+  {:font-size       14
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: Why extra spaces?

File: src/status_im/ui/screens/wallet/transactions/subs.cljs [hunk 2]
```diff
@@ -57,3 +62,16 @@
...
+  (fn [transactions]
+    (let [{:keys [gas-used gas-price timestamp] :as transaction} (first (:inbound transactions))]
+      (assoc transaction
+             :cost (money/wei->ether (money/fee-value gas-used gas-price))
+             :date (datetime/timestamp->long-date timestamp)))))
+
+(reg-sub :wallet.transactions.details/confirmations
+  :<- [:wallet.transactions/transaction-details]
+  (fn [transaction-details]
+    3 #_(:confirmations transaction-details)))
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/subs.cljs]: WIP ?

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 3]
```diff
@@ -50,32 +56,36 @@
...
+    :unsigned               (transaction-icon :icons/dots-horizontal styles/color-gray4-transparent styles/color-gray7)
+    :inbound                (transaction-icon :icons/arrow-left styles/color-green-3-light styles/color-green-3)
+    :outbound               (transaction-icon :icons/arrow-right styles/color-blue4-transparent styles/color-blue4)
+    (:postponed :pending)   (transaction-icon :icons/arrow-right styles/color-gray4-transparent styles/color-gray7)
+    (throw (str "Unknown transaction type: " k))))
 
 (defn render-transaction [{:keys [to from type value symbol] :as m}]
   [list/item
-   [list/item-icon (transaction-type->icon type)]
+   [list/item-icon (transaction-type->icon (keyword type))]
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: Why this change?

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 4]
```diff
@@ -192,3 +202,59 @@
...
+    [react/view {:style (transactions.styles/transaction-details-confirmations-progress-bar-progress confirmations)}]
+    [react/view {:style (transactions.styles/transaction-details-confirmations-progress-bar-total confirmations)}]]
+   [react/text {:style transactions.styles/transaction-details-confirmations-count}
+    (str confirmations " " (i18n/label :t/confirmations))]
+   [react/text {:style transactions.styles/transaction-details-confirmations-helper-text}
+    (i18n/label :t/confirmations-helper-text)]])
+
+(defn transaction-details-list [{:keys [block hash from to gas-limit gas-price gas-used cost nonce data]}]
+  [react/view {:style transactions.styles/transaction-details-block}
+   [transaction-details-list-row :t/block block]
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: Would it be worth having a higher level component that would take a map? Can this component be reused in some other screen?

File: src/status_im/ui/screens/wallet/transactions/styles.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/styles.cljs]: removed longer element and forgot to clean up

File: src/status_im/ui/screens/wallet/transactions/subs.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/subs.cljs]: yes for now design is ready but some work is required on the data-binding side

File: src/status_im/ui/screens/wallet/transactions/views.cljs (hunk 3 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/views.cljs]: in order to reuse it for transaction-details as well, and keywords is more idiomatic

File: src/status_im/ui/screens/wallet/transactions/views.cljs (hunk 4 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/transactions/views.cljs]: hard to say regarding higher level component because there will be some edge cases, it's not fully complete yet since from/to for instance should show the wallet in black and address in gray for the known addresses.

I haven't seen an other screen that needs such compoment

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 5]
```diff
@@ -15,7 +15,9 @@
    :padding-right 10
    :font-size     13})
 
-(def main-section styles/flex)
+(def main-section
+  {:flex 1
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: indentation

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 6]
```diff
@@ -88,3 +90,80 @@
...
+   :margin-right 10
+   :color        styles/color-gray4
+   :font-size    14})
+
+(def transaction-details-item-value-wrapper
+  {:flex 5})
+
+(def transaction-details-item-value
+  {:font-size 14
+   :color styles/color-black})
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: indentation

File: src/status_im/ui/screens/wallet/transactions/styles.cljs [hunk 7]
```diff
@@ -88,3 +90,80 @@
...
+
+(def transaction-details-item-extra-value
+  {:font-size 14
+   :color styles/color-gray4})
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/styles.cljs]: indentation

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 8]
```diff
@@ -9,12 +9,18 @@
             [status-im.components.styles :as styles]
             [status-im.components.tabs.views :as tabs]
             [status-im.components.toolbar-new.view :as toolbar]
+            [status-im.components.toolbar-new.actions :as actions]
             [status-im.i18n :as i18n]
             [status-im.ui.screens.wallet.transactions.styles :as transactions.styles]
             [status-im.ui.screens.wallet.views :as wallet.views]
-            [status-im.utils.utils :as utils])
+            [status-im.utils.utils :as utils]
+            [status-im.utils.hex :as i]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: i?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

**Review by flexsurfer — APPROVED** [review]
